spec/globus_client_spec.rb

```diff
@@ -98,16 +98,47 @@
   end
 
   describe "#user_exists?" do
-    let(:fake_identity) { instance_double(described_class::Identity, exists?: nil) }
+    context "when request is successful" do
+      let(:fake_identity) { instance_double(described_class::Identity, exists?: nil) }
 
-    before do
-      allow(described_class::Identity).to receive(:new).and_return(fake_identity)
-      allow(fake_identity).to receive(:exists?)
+      before do
+        allow(described_class::Identity).to receive(:new).and_return(fake_identity)
+        allow(fake_identity).to receive(:exists?)
+      end
+
+      it "invokes Identity#exists?" do
+        client.user_exists?(sunetid: "bogus")
+        expect(fake_identity).to have_received(:exists?).once
+      end
     end
 
-    it "invokes Identity#exists?" do
-      client.user_exists?(sunetid: "bogus")
-      expect(fake_identity).to have_received(:exists?).once
+    context "when token is expired" do
+      let(:fake_identity) { instance_double(described_class::Identity, exists?: nil) }
+
+      before do
+        allow(described_class::Identity).to receive(:new).and_return(fake_identity)
+        allow(GlobusClient::Authenticator).to receive(:token).and_return("a_token", "new_token")
+        allow(fake_identity).to receive(:exists?).once.and_raise(GlobusClient::UnexpectedResponse::UnauthorizedError)
+        allow(fake_identity).to receive(:exists?).once.and_return("user-identity-uuid")
+      end
+
+      it "retries Identity#exists?" do
+        expect(client.user_exists?(sunetid: "user")).to eq("user-identity-uuid")
```

Comment:
Maybe this is the Monday morning brain speaking, but doesn't the `exists?` method return a boolean instead of an ID? Prolly just a stubbing thing (changing line 122 to match the behavior of the method)?

Reply:
Thanks for catching this. I'm the one with Monday morning brain. :) Fixed to return `true`.